test/erlfmt_markdown_SUITE.erl

```diff
@@ -0,0 +1,175 @@
+%% limitations under the License.
+-module(erlfmt_markdown_SUITE).
+
+-include_lib("common_test/include/ct.hrl").
+
+-include_lib("stdlib/include/assert.hrl").
+
+%% Test server callbacks
+-export([
+    suite/0,
+    all/0,
+    groups/0,
+    init_per_suite/1,
+    end_per_suite/1,
+    init_per_group/2,
+    end_per_group/2,
+    init_per_testcase/2,
+    end_per_testcase/2
+]).
+
+%% Test cases
+-export([
+    markdown_files/1,
+    markdown_string/1
+]).
+
+suite() ->
+    [{timetrap, {seconds, 10}}].
+
+init_per_suite(Config) ->
+    Config.
+
+end_per_suite(_Config) ->
+    ok.
+
+init_per_group(_GroupName, Config) ->
+    Config.
+
+end_per_group(_GroupName, _Config) ->
+    ok.
+
+init_per_testcase(_TestCase, Config) ->
+    Config.
+
+end_per_testcase(_TestCase, _Config) ->
+    ok.
+
+groups() ->
+    [
+        {markdown, [parallel], [
+            markdown_files,
+            markdown_string
+        ]}
+    ].
+
+all() ->
+    [{group, markdown}].
+
+markdown_files(Config) when is_list(Config) ->
+    {ok, Data} = file:get_cwd(),
+    RepoRootPath = filename:join([Data, "..", "..", "..", ".."]),
+    DocPath = filename:join(RepoRootPath, "doc"),
+    Filenames = lists:append(
+        find_markdown_filenames_in(RepoRootPath),
+        find_markdown_filenames_in(DocPath)
+    ),
+    ?assertMatch([_One | [_Two | [_Three | _AtLeast]]], Filenames),
+    lists:map(
+        fun (Filename) ->
+            {ok, Content} = file:read_file(Filename),
+            check_markdown(binary_to_list(Content))
+        end,
+        Filenames
+    ).
+
+find_markdown_filenames_in(Path) ->
+    {ok, BaseFilenames} = file:list_dir_all(Path),
+    Filenames =
+        lists:map(fun (Filename) -> filename:join([Path, Filename]) end, BaseFilenames),
+    lists:filter(fun (Filename) -> filename:extension(Filename) == ".md" end, Filenames).
+
+markdown_string(Config) when is_list(Config) ->
+    S =
+        "# Heading\n"
+        "ignore for now\n"
+        "```erlang formatted key\n"
+        "hello(mike, joe, robert)\n"
+        "```\n"
+        "ignore for now\n"
+        "```erlang unformatted\n"
+        "goodbye(mike, joe, robert)\n"
+        "```\n"
+        "The previous unformatted version, should match this formatted version\n"
+        "```erlang formatted\n"
+        "goodbye(mike, joe, robert)\n"
+        "```\n"
+        "This unformatted version, should match the first formatted version with the same key\n"
+        "```erlang unformatted key\n"
+        "hello(mike, joe, robert)\n"
+        "```\n",
+    check_markdown(S).
+
+check_markdown(Content) ->
+    Sections = string:split(Content, "```", all),
+    lists:foldl(fun check_section/2, {text, maps:new()}, Sections).
+
+check_section(_Text, {text, State}) ->
+    {code, State};
+check_section(Text, {code, State}) ->
+    [FirstLine, CurrentCode] = string:split(Text, "\n"),
+    Spec = string:split(FirstLine, " ", all),
+    case Spec of
+        [Erlang | [ToFormat | Keys]] ->
```

Comment:
This could be simplified and combined with the following prefix check using something like:
```erl
["erl" ++ _, ToFormat | Keys]
```

Reply:
I was looking for this, thank you